src/main/kotlin/fi/metatavu/oss/api/impl/surveys/SurveyController.kt

```diff
@@ -19,9 +20,22 @@ class SurveyController {
      *
      * @param firstResult first result
      * @param maxResults max results
+     * @param status status
      * @return uni with list of surveys and count
      */
-    suspend fun listSurveys(firstResult: Int?, maxResults: Int?): Pair<List<SurveyEntity>, Long> {
+    suspend fun listSurveys(
+        firstResult: Int?,
+        maxResults: Int?,
+        status: SurveyStatus?
+    ): Pair<List<SurveyEntity>, Long> {
+        if (status != null) {
+            return surveyRepository.applyPagingToQuery(
```

Comment:
Shouldn't applyPagingToQuery be used internally by the repository?

Reply:
Resolved.